docs/mine-hnt/validators/setup.mdx

```diff
@@ -0,0 +1,161 @@
+
+```sudo usermod -aG docker $USER```
+
+Log in and out of your account to apply these changes. You are now ready to use Docker.
+
+### Run a docker container
+
+Note: on amd64 systems AVX support is required. Verify that it exists on your host system:
+```
+grep avx /proc/cpuinfo
+grep avx2 /proc/cpuinfo
+```
+If nothing is returned from these commands, your host system does not have AVX support and your validator may not be stable.
+
+### Start container
+
+Before running the container for the first time, it is advisable to pick a "system mount point", i.e., a directory in your host system; some long-term data is stored there. This allows you to easily maintain your validator's blockchain identity (i.e. swarm keys) and blockchain state through image updates.
+
+If you are using a Linux system, you can just create a directory in your user's home directory:
+```mkdir ~/miner_data```
+
+If you are using Ubuntu as user ubuntu, this path would now be ```/home/ubuntu/miner_data```. This will be used later.
+Now you can try the ```run``` command to start your container for the first time:
+
+```
+docker run -d \
+--restart always \
+--publish 2154:2154/tcp \
+--name miner \
+--mount type=bind,source=/home/ubuntu/miner_data,target=/var/data \
+quay.io/team-helium/miner:miner-xxxNN_YYYY.MM.DD
+```
+
+Replace ```xxxNN``` with the architecture used, i.e. amd64 or arm64, and with the release date of the image.
+
+The ```-d``` option runs in detached mode, which makes the command return or not; you may want to omit if you have a daemon manager running the docker for you.
+
+The ```--restart always``` option asks Docker to keep the image running, starting the image on boot and restarting the image if it crashes. Depending on how you installed Docker in your system, it'll start on boot. In the AWS AMI above, we use systemd (```systemctl status docker``` to check).
+
+The ```--name miner``` names the container, which makes interacting with the docker easier, but feel free to name the container whatever you want.
+
+The ```--mount``` with the parameters above will mount the container's ```/var/data/``` directory to the systems directory ```/home/ec2-user/miner_data```.
+
+### Interact with the validator within the container
+
+You may want to interrogate the Validator or interact with it. Docker's ```exec``` command enables this e.g.:
+```docker exec miner miner info height```
+
+In other words, prepend ```docker exec miner``` to any of the commands documented or create an alias such as: ```alias miner="docker exec miner miner"```.
+And start the container again as described above, but with the new release tag. 
+
+Thanks to the ```--mount``` option, the blockchain data and the validator keys are preserved through updates.
+
+After setup is complete, check [Monitor a validator section](/mine-hnt/validators/validator-monitor).
+
+## <a name="source"></a>From source
```

Comment:
I think @amirhaleem was going to take a stab here but I can come back to it if needed.

Reply:
you mean testing source e2e